### PR TITLE
Postprocessor dependencies

### PR DESCRIPTION
**Pull request by yan12125**
Try to solve #9073 by introducing "Download targets" and check them in `YoutubeDL.post_process()`.

### CODE REVIEW

File: youtube_dl/YoutubeDL.py [hunk 1]
```diff
@@ -1563,8 +1569,12 @@ def process_info(self, info_dict):
                             subfile.write(sub_data)
                 except (OSError, IOError):
                     self.report_error('Cannot write subtitles file ' + sub_filename)
+                    all_subtitles_ok = False
                     return
```
Comment (yan12125) [inline youtube_dl/YoutubeDL.py]: This ```return``` is introduced in a0432a1e8086e2318b122744c7afa5145abdf663. I guess failures in downloading subtitles should be non-fatal, just like thumbnails?

File: youtube_dl/YoutubeDL.py [hunk 2]
```diff
@@ -48,6 +48,7 @@
     determine_ext,
     determine_protocol,
     DownloadError,
+    DownloadTarget,
```
Comment (dstftw) [inline youtube_dl/YoutubeDL.py]: Probably an `asset` fits better to denote the notion.

File: youtube_dl/YoutubeDL.py [hunk 3]
```diff
@@ -1462,6 +1463,7 @@ def process_info(self, info_dict):
         self._num_downloads += 1
 
         info_dict['_filename'] = filename = self.prepare_filename(info_dict)
+        info_dict['_downloaded_targets'] = []
```
Comment (dstftw) [inline youtube_dl/YoutubeDL.py]: It's probably makes sense to remove it from resulting output JSON.

File: youtube_dl/__init__.py [hunk 4]
```diff
@@ -252,7 +252,7 @@ def parse_retries(retries):
             'key': 'EmbedThumbnail',
             'already_have_thumbnail': already_have_thumbnail
         })
-        if not already_have_thumbnail:
+        if not already_have_thumbnail and not opts.skip_download:
```
Comment (dstftw) [inline youtube_dl/__init__.py]: This changes the behavior of the following use case:
1. Media was downloaded previously (file exists).
2. youtube-dl is run with `--embed-thumbnail --skip-download`.
3. No thumbnail is downloaded and embedded (it does both with current code).
Is this intentional to match `--embed-thumbnail --skip-download` behavior?

File: youtube_dl/YoutubeDL.py (hunk 1 above)
Comment (dstftw) [inline youtube_dl/YoutubeDL.py]: Sounds reasonable.

File: youtube_dl/YoutubeDL.py (hunk 2 above)
Comment (yan12125) [inline youtube_dl/YoutubeDL.py]: Did you mean naming the class as ```Asset```?

File: youtube_dl/__init__.py (hunk 4 above)
Comment (yan12125) [inline youtube_dl/__init__.py]: Thanks. Will change this.

**Pull request closed without merging**